# Web Scraping

**Web scraping del link de Wikipedia donde aparece el listado de películas ganadoras y nominadas cada año en la categoría "Oscar a la mejor película"**

In [163]:
import requests
from bs4 import BeautifulSoup
import pandas as pd

url = 'https://en.wikipedia.org/wiki/List_of_Academy_Award-winning_films'

html = requests.get(url).content
soup = BeautifulSoup(html, "lxml")
soup

<!DOCTYPE html>
<html class="client-nojs" dir="ltr" lang="en">
<head>
<meta charset="utf-8"/>
<title>List of Academy Award-winning films - Wikipedia</title>
<script>document.documentElement.className="client-js";RLCONF={"wgBreakFrames":!1,"wgSeparatorTransformTable":["",""],"wgDigitTransformTable":["",""],"wgDefaultDateFormat":"dmy","wgMonthNames":["","January","February","March","April","May","June","July","August","September","October","November","December"],"wgMonthNamesShort":["","Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"],"wgRequestId":"XkbeJgpAAEMAACTHWjQAAAAI","wgCSPNonce":!1,"wgCanonicalNamespace":"","wgCanonicalSpecialPageName":!1,"wgNamespaceNumber":0,"wgPageName":"List_of_Academy_Award-winning_films","wgTitle":"List of Academy Award-winning films","wgCurRevisionId":940793266,"wgRevisionId":940793266,"wgArticleId":3578923,"wgIsArticle":!0,"wgIsRedirect":!1,"wgAction":"view","wgUserName":null,"wgUserGroups":["*"],"wgCategories":["Articles with shor

**Extraemos aquellos resultados que están en el URL sombreados en amarillo, que son las películas ganadoras**

In [164]:
row_winners = soup.find_all('tr', {'style' : 'background:#EEDD82'})
row_winners

[<tr style="background:#EEDD82">
 <td><i><b><a href="/wiki/Parasite_(2019_film)" title="Parasite (2019 film)">Parasite</a></b></i></td>
 <td><a href="/wiki/2019_in_film" title="2019 in film">2019</a></td>
 <td>4</td>
 <td>6
 </td></tr>, <tr style="background:#EEDD82">
 <td><i><b><a href="/wiki/Green_Book_(film)" title="Green Book (film)">Green Book</a></b></i></td>
 <td><a href="/wiki/2018_in_film" title="2018 in film">2018</a></td>
 <td>3</td>
 <td>5
 </td></tr>, <tr style="background:#EEDD82">
 <td><i><b><a href="/wiki/The_Shape_of_Water" title="The Shape of Water">The Shape of Water</a></b></i></td>
 <td><a href="/wiki/2017_in_film" title="2017 in film">2017</a></td>
 <td>4</td>
 <td>13
 </td></tr>, <tr style="background:#EEDD82">
 <td><i><b><a href="/wiki/Moonlight_(2016_film)" title="Moonlight (2016 film)">Moonlight</a></b></i></td>
 <td><a href="/wiki/2016_in_film" title="2016 in film">2016</a></td>
 <td>3</td>
 <td>8
 </td></tr>, <tr style="background:#EEDD82">
 <td><i><b><a hre

**Extraemos los elementos de texto de cada línea y hacemos split por cada salto de línea.**

In [165]:
row_winners = [row.text.strip().split("\n") for row in row_winners]
row_winners

[['Parasite', '2019', '4', '6'],
 ['Green Book', '2018', '3', '5'],
 ['The Shape of Water', '2017', '4', '13'],
 ['Moonlight', '2016', '3', '8'],
 ['Spotlight', '2015', '2', '6'],
 ['Birdman', '2014', '4', '9'],
 ['12 Years a Slave', '2013', '3', '9'],
 ['Argo', '2012', '3', '7'],
 ['The Artist', '2011', '5', '10'],
 ["The King's Speech", '2010', '4', '12'],
 ['The Hurt Locker', '2009', '6', '9'],
 ['Slumdog Millionaire', '2008', '8', '10'],
 ['No Country for Old Men', '2007', '4', '8'],
 ['The Departed', '2006', '4', '5'],
 ['Crash', '2005', '3', '6'],
 ['Million Dollar Baby', '2004', '4', '7'],
 ['The Lord of the Rings: The Return of the King', '2003', '11', '11'],
 ['Chicago', '2002', '6', '13'],
 ['A Beautiful Mind', '2001', '4', '8'],
 ['Gladiator', '2000', '5', '12'],
 ['American Beauty', '1999', '5', '8'],
 ['Shakespeare in Love', '1998', '7', '13'],
 ['Titanic', '1997', '11', '14'],
 ['The English Patient', '1996', '9', '12'],
 ['Braveheart', '1995', '5', '10'],
 ['Forrest Gump

**Nos queda como resultado una lista de listas con cada título de película, su año, Oscars ganados, y nominaciones**

**Convertimos a DataFrame.**

**Cambiamos el nombre de las columnas para identificarlas mejor.**

**Reemplazamos algunos valores que están introducidos de manera poco precisa en el dataset"

In [166]:
pd.set_option('display.max_rows', 200)

colnames = ['Title','Year','Awards', 'Nominations']

df_movies_wiki = pd.DataFrame(row_winners[:], columns = colnames)

df_movies_wiki = df_movies_wiki[['Year','Title','Awards','Nominations']]

df_movies_wiki['Year'] = df_movies_wiki['Year'].replace('1932/33', '1933')
df_movies_wiki['Year'] = df_movies_wiki['Year'].replace('1931/32', '1932')
df_movies_wiki['Year'] = df_movies_wiki['Year'].replace('1930/31', '1931')
df_movies_wiki['Year'] = df_movies_wiki['Year'].replace('1929/30', '1930')
df_movies_wiki['Year'] = df_movies_wiki['Year'].replace('1928/29', '1929')
df_movies_wiki['Year'] = df_movies_wiki['Year'].replace('1927/28', '1928')
df_movies_wiki['Year'].str.strip()

df_movies_wiki['Awards'] = df_movies_wiki['Awards'].replace('5 (1)', '6')
df_movies_wiki['Awards'] = df_movies_wiki['Awards'].replace('7 (1)', '8')
df_movies_wiki['Awards'] = df_movies_wiki['Awards'].replace('8 (2)', '10')

df_movies_wiki['Nominations'] = df_movies['Nominations'].replace('10[4]', '10')

df_movies_wiki.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 92 entries, 0 to 91
Data columns (total 4 columns):
Year           92 non-null object
Title          92 non-null object
Awards         92 non-null object
Nominations    92 non-null int64
dtypes: int64(1), object(3)
memory usage: 3.0+ KB


**Convertimos las columnas que tienen números a datos numéricos**

In [167]:
df_movies_wiki['Year'] = pd.to_numeric(df_movies_wiki['Year'], errors='coerce')
df_movies_wiki['Awards'] = pd.to_numeric(df_movies_wiki['Awards'], errors='coerce')
df_movies_wiki['Nominations'] = pd.to_numeric(df_movies_wiki['Nominations'], errors='coerce')

df_movies_wiki.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 92 entries, 0 to 91
Data columns (total 4 columns):
Year           92 non-null int64
Title          92 non-null object
Awards         92 non-null int64
Nominations    92 non-null int64
dtypes: int64(3), object(1)
memory usage: 3.0+ KB


**Ordenamos de más reciente a más antigua.**

In [168]:
df_movies_wiki.sort_values(by='Year', ascending=False)

df_movies_wiki

Year                                          Title  Awards  Nominations
0   2019                                       Parasite       4            6
1   2018                                     Green Book       3            5
2   2017                             The Shape of Water       4           13
3   2016                                      Moonlight       3            8
4   2015                                      Spotlight       2            6
5   2014                                        Birdman       4            9
6   2013                               12 Years a Slave       3            9
7   2012                                           Argo       3            7
8   2011                                     The Artist       5           10
9   2010                              The King's Speech       4           12
10  2009                                The Hurt Locker       6            9
11  2008                            Slumdog Millionaire       8           10
12  2007                         No Country for Old Men       4            8
13  2006                                   The Departed       4            5
14  2005                                          Crash       3            6
15  2004                            Million Dollar Baby       4            7
16  2003  The Lord of the Rings: The Return of the King      11           11
17  2002                                        Chicago       6           13
18  2001                               A Beautiful Mind       4            8
19  2000                                      Gladiator       5           12
20  1999                                American Beauty       5            8
21  1998                            Shakespeare in Love       7           13
22  1997                                        Titanic      11           14
23  1996                            The English Patient       9           12
24  1995                                     Braveheart       5           10
25  1994                                   Forrest Gump       6           13
26  1993                               Schindler's List       7           12
27  1992                                     Unforgiven       4            9
28  1991                       The Silence of the Lambs       5            7
29  1990                             Dances with Wolves       7           12
30  1989                             Driving Miss Daisy       4            9
31  1988                                       Rain Man       4            8
32  1987                               The Last Emperor       9            9
33  1986                                        Platoon       4            8
34  1985                                  Out of Africa       7           11
35  1984                                        Amadeus       8           11
36  1983                            Terms of Endearment       5           11
37  1982                                         Gandhi       8           11
38  1981                               Chariots of Fire       4            7
39  1980                                Ordinary People       4            6
40  1979                              Kramer vs. Kramer       5            9
41  1978                                The Deer Hunter       5            9
42  1977                                     Annie Hall       4            5
43  1976                                          Rocky       3           10
44  1975                One Flew over the Cuckoo's Nest       5            9
45  1974                          The Godfather Part II       6           11
46  1973                                      The Sting       7           10
47  1972                                  The Godfather       3           10
48  1971                          The French Connection       5            8
49  1970                                         Patton       7           10
50  1969                                Midnight Cowboy       3            7


**Ordenamos por número de Oscars recibidos.**

In [172]:
df_movies_wiki.sort_values(by='Awards', ascending=False).reset_index(drop=True).head(10)

Year                                          Title  Awards  Nominations
0  2003  The Lord of the Rings: The Return of the King      11           11
1  1997                                        Titanic      11           14
2  1959                                        Ben-Hur      11           12
3  1961                                West Side Story      10           11
4  1939                             Gone with the Wind      10           13
5  1958                                           Gigi       9            9
6  1987                               The Last Emperor       9            9
7  1996                            The English Patient       9           12
8  1984                                        Amadeus       8           11
9  1946                    The Best Years of Our Lives       8            8

**Ordenamos por número de nominaciones recibidas.**

In [174]:
df_movies_wiki.sort_values(by='Nominations', ascending=False).reset_index(drop=True).head(10)

Year                  Title  Awards  Nominations
0  1950          All About Eve       6           14
1  1997                Titanic      11           14
2  1953  From Here to Eternity       8           13
3  1994           Forrest Gump       6           13
4  2002                Chicago       6           13
5  1998    Shakespeare in Love       7           13
6  1939     Gone with the Wind      10           13
7  2017     The Shape of Water       4           13
8  1942           Mrs. Miniver       6           12
9  2010      The King's Speech       4           12

**Ratio de nominaciones/oscars ganados**

In [176]:
df_movies_wiki['Ratio Nominations/Awards'] = round((df_movies_wiki['Awards'] / df_movies_wiki['Nominations']) * 100)
df_movies_wiki.sort_values(by = 'Ratio Nominations/Awards', ascending = False).reset_index(drop=True).head(10)

Year                                          Title  Awards  Nominations  \
0  1928                                          Wings       2            2   
1  1946                    The Best Years of Our Lives       8            8   
2  1934                          It Happened One Night       5            5   
3  1932                                    Grand Hotel       1            1   
4  2003  The Lord of the Rings: The Return of the King      11           11   
5  1987                               The Last Emperor       9            9   
6  1958                                           Gigi       9            9   
7  1959                                        Ben-Hur      11           12   
8  1961                                West Side Story      10           11   
9  1957                   The Bridge on the River Kwai       7            8   

   Ratio Nominations/Awards  
0                     100.0  
1                     100.0  
2                     100.0  
3                     100.0  
4                     100.0  
5                     100.0  
6                     100.0  
7                      92.0  
8                      91.0  
9                      88.0

Me parecía curioso destacar cuáles han sido las películas que han tenido un mejor ratio de nominaciones/premios ganados. 

Destaco "Lord of The Rings", que ganó los 11 premios a los que estaba nominado. Así como "The Last Emperor" o "Gigi" que ganaron 9 de 9. Espectacular.

No al 100% pero un gran ratio y número de premios fueron "Ben-Hur" y "West Side Story" que ganaron 11 y 10 de los 12 y 11 que estaban nominados respectivamente.

# API

**Usaremos la API de OMDB, un portal especializado en cine donde nos dan información sobre películas.**

Hacemos un primer ejemplo seleccionando una película cualquiera, como "Parasite" para ver cómo funciona la API y qué información nos proporciona sobre la película.

In [177]:
import json
import requests


url = "http://www.omdbapi.com/?t=parasite&plot=full&apikey=e8d07dc2"

response = requests.get(url)

results = response.json()

results

{'Title': 'Parasite',
 'Year': '2019',
 'Rated': 'R',
 'Released': '08 Nov 2019',
 'Runtime': '132 min',
 'Genre': 'Comedy, Crime, Drama, Thriller',
 'Director': 'Bong Joon Ho',
 'Writer': 'Bong Joon Ho (story), Bong Joon Ho (screenplay), Jin Won Han (screenplay)',
 'Actors': 'Kang-ho Song, Sun-kyun Lee, Yeo-jeong Jo, Woo-sik Choi',
 'Plot': 'All unemployed, Ki-taek and his family take peculiar interest in the wealthy and glamorous Parks, as they ingratiate themselves into their lives and get entangled in an unexpected incident.',
 'Language': 'Korean, English',
 'Country': 'South Korea',
 'Awards': 'Nominated for 6 Oscars. Another 199 wins & 183 nominations.',
 'Poster': 'https://m.media-amazon.com/images/M/MV5BYWZjMjk3ZTItODQ2ZC00NTY5LWE0ZDYtZTI3MjcwN2Q5NTVkXkEyXkFqcGdeQXVyODk4OTc3MTY@._V1_SX300.jpg',
 'Ratings': [{'Source': 'Internet Movie Database', 'Value': '8.6/10'},
  {'Source': 'Rotten Tomatoes', 'Value': '99%'},
  {'Source': 'Metacritic', 'Value': '96/100'}],
 'Metascore': '96

**Convertimos a DataFrame para tener una manera más fácil y útil de tener la información**

In [178]:
data = pd.DataFrame(results)
data.head(1)

Title  Year Rated     Released  Runtime                           Genre  \
0  Parasite  2019     R  08 Nov 2019  132 min  Comedy, Crime, Drama, Thriller   

       Director                                             Writer  \
0  Bong Joon Ho  Bong Joon Ho (story), Bong Joon Ho (screenplay...   

                                              Actors  \
0  Kang-ho Song, Sun-kyun Lee, Yeo-jeong Jo, Woo-...   

                                                Plot  ... Metascore  \
0  All unemployed, Ki-taek and his family take pe...  ...        96   

  imdbRating imdbVotes     imdbID   Type          DVD BoxOffice Production  \
0        8.6   193,674  tt6751668  movie  14 Jan 2020       N/A       NEON   

  Website Response  
0     N/A     True  

[1 rows x 25 columns]

**Vamos a llamar a esta API con todas las películas que hemos sacado como ganadoras de Oscar a Mejor Película en la primera parte del ejercicio. Creamos una lista con todas ellas e iteramos para llamar a la API por cada una de ellas**

In [179]:
movies_list = df_movies['Title'].tolist()
print(movies_list)

['Parasite', 'Green Book', 'The Shape of Water', 'Moonlight', 'Spotlight', 'Birdman', '12 Years a Slave', 'Argo', 'The Artist', "The King's Speech", 'The Hurt Locker', 'Slumdog Millionaire', 'No Country for Old Men', 'The Departed', 'Crash', 'Million Dollar Baby', 'The Lord of the Rings: The Return of the King', 'Chicago', 'A Beautiful Mind', 'Gladiator', 'American Beauty', 'Shakespeare in Love', 'Titanic', 'The English Patient', 'Braveheart', 'Forrest Gump', "Schindler's List", 'Unforgiven', 'The Silence of the Lambs', 'Dances with Wolves', 'Driving Miss Daisy', 'Rain Man', 'The Last Emperor', 'Platoon', 'Out of Africa', 'Amadeus', 'Terms of Endearment', 'Gandhi', 'Chariots of Fire', 'Ordinary People', 'Kramer vs. Kramer', 'The Deer Hunter', 'Annie Hall', 'Rocky', "One Flew over the Cuckoo's Nest", 'The Godfather Part II', 'The Sting', 'The Godfather', 'The French Connection', 'Patton', 'Midnight Cowboy', 'Oliver!', 'In the Heat of the Night', 'A Man for All Seasons', 'The Sound of Mu

In [184]:
url = "http://www.omdbapi.com"

output_results = []

for title in movies_list:
    response = requests.get(url, params={"t":title, "apikey": "e8d07dc2"})
    output_results.append(response.json())

df_movies_API = pd.DataFrame(output_results)
df_movies_API.head()

Title  Year  Rated     Released  Runtime  \
0            Parasite  2019      R  08 Nov 2019  132 min   
1          Green Book  2018  PG-13  16 Nov 2018  130 min   
2  The Shape of Water  2017      R  22 Dec 2017  123 min   
3           Moonlight  2016      R  18 Nov 2016  111 min   
4           Spotlight  2015      R  20 Nov 2015  129 min   

                                          Genre            Director  \
0                Comedy, Crime, Drama, Thriller        Bong Joon Ho   
1               Biography, Comedy, Drama, Music      Peter Farrelly   
2  Adventure, Drama, Fantasy, Romance, Thriller  Guillermo del Toro   
3                                         Drama       Barry Jenkins   
4                       Biography, Crime, Drama        Tom McCarthy   

                                              Writer  \
0  Bong Joon Ho (story), Bong Joon Ho (screenplay...   
1  Nick Vallelonga, Brian Hayes Currie, Peter Far...   
2  Guillermo del Toro (screenplay by), Vanessa Ta...   
3  Barry Jenkins (screenplay by), Tarell Alvin Mc...   
4                          Josh Singer, Tom McCarthy   

                                              Actors  \
0  Kang-ho Song, Sun-kyun Lee, Yeo-jeong Jo, Woo-...   
1  Viggo Mortensen, Mahershala Ali, Linda Cardell...   
2  Sally Hawkins, Michael Shannon, Richard Jenkin...   
3  Mahershala Ali, Shariff Earp, Duan Sanderson, ...   
4  Mark Ruffalo, Michael Keaton, Rachel McAdams, ...   

                                                Plot  ... imdbRating  \
0  All unemployed, Ki-taek and his family take pe...  ...        8.6   
1  A working-class Italian-American bouncer becom...  ...        8.2   
2  At a top secret research facility in the 1960s...  ...        7.3   
3  A young African-American man grapples with his...  ...        7.4   
4  The true story of how the Boston Globe uncover...  ...        8.1   

  imdbVotes     imdbID   Type          DVD    BoxOffice  \
0   193,674  tt6751668  movie  14 Jan 2020          N/A   
1   290,955  tt6966692  movie  19 Feb 2019          N/A   
2   338,128  tt5580390  movie  13 Mar 2018  $57,393,976   
3   246,764  tt4975722  movie  28 Feb 2017  $27,850,910   
4   385,890  tt1895587  movie  23 Feb 2016          N/A   

                 Production Website Response totalSeasons  
0                      NEON     N/A     True          NaN  
1        Universal Pictures     N/A     True          NaN  
2  Fox Searchlight Pictures     N/A     True          NaN  
3                 A24 Films     N/A     True          NaN  
4           Open Road Films     N/A     True          NaN  

[5 rows x 26 columns]

**Eliminamos del DataFrame algunas columnas que no nos son relevantes para este ejercicio**

In [185]:
df_movies_API = df_movies_API.drop(columns = ['Rated', 'Released','Writer','Plot','Language', 'Poster','Awards','BoxOffice','Ratings','Metascore','imdbVotes','imdbID', 'Year','Type','DVD','Production','Website','totalSeasons','Response'])
df_movies_API

Title  Runtime  \
0                                        Parasite  132 min   
1                                      Green Book  130 min   
2                              The Shape of Water  123 min   
3                                       Moonlight  111 min   
4                                       Spotlight  129 min   
5                                         Birdman   30 min   
6                                12 Years a Slave  134 min   
7                                            Argo  120 min   
8                                      The Artist  100 min   
9                               The King's Speech  118 min   
10                                The Hurt Locker  131 min   
11                            Slumdog Millionaire  120 min   
12                         No Country for Old Men  122 min   
13                                   The Departed  151 min   
14                                          Crash  112 min   
15                            Million Dollar Baby  132 min   
16  The Lord of the Rings: The Return of the King  201 min   
17                                        Chicago  113 min   
18                               A Beautiful Mind  135 min   
19                                      Gladiator  155 min   
20                                American Beauty  122 min   
21                            Shakespeare in Love  123 min   
22                                        Titanic  194 min   
23                            The English Patient  162 min   
24                                     Braveheart  178 min   
25                                   Forrest Gump  142 min   
26                               Schindler's List  195 min   
27                                     Unforgiven  130 min   
28                       The Silence of the Lambs  118 min   
29                             Dances with Wolves  181 min   
30                             Driving Miss Daisy   99 min   
31                                       Rain Man  133 min   
32                               The Last Emperor  163 min   
33                                        Platoon  120 min   
34                                  Out of Africa  161 min   
35                                        Amadeus  160 min   
36                            Terms of Endearment  132 min   
37                                         Gandhi  191 min   
38                               Chariots of Fire  125 min   
39                                Ordinary People  124 min   
40                              Kramer vs. Kramer  105 min   
41                                The Deer Hunter  183 min   
42                                     Annie Hall   93 min   
43                                          Rocky  120 min   
44                One Flew Over the Cuckoo's Nest  133 min   
45                         The Godfather: Part II  202 min   
46                                      The Sting  129 min   
47                                  The Godfather  175 min   
48                          The French Connection  104 min   
49                                         Patton  172 min   
50                                Midnight Cowboy  113 min   
51                                        Oliver!  153 min   
52                       In the Heat of the Night  110 min   
53                          A Man for All Seasons  120 min   
54                             The Sound of Music  172 min   
55                                  All About Eve  138 min   
56                 All Quiet on the Western Front  136 min   
57                             All the King's Men  128 min   
58                           An American in Paris  114 min   
59                                  The Apartment  125 min   
60                    Around the World in 80 Days  120 min   
61                                        Ben-Hur  212 min   
62                    The Best Years of Our Lives  170 min   
63                   The Bridge on the River Kwai  161 min   
64            

# Combinación Web Scraping + APIs

**Hacemos un merge entre los dos DataFrame que hemos sacado. Por un lado el de Web Scraping y por otro lado el de API.**
**Relacionaremos así los datos obtenidos de cada una de las fuentes y tendremos una mejor visión para analizar cuál ha sido la puntuación recibida en IMDB cada película ganadora del Oscar a la mejor Película cada año**

In [186]:
df_movies_final = pd.merge(df_movies_wiki, df_movies_API, on='Title')
df_movies_final

Year                                          Title  Awards  Nominations  \
0   2019                                       Parasite       4            6   
1   2018                                     Green Book       3            5   
2   2017                             The Shape of Water       4           13   
3   2016                                      Moonlight       3            8   
4   2015                                      Spotlight       2            6   
5   2014                                        Birdman       4            9   
6   2013                               12 Years a Slave       3            9   
7   2012                                           Argo       3            7   
8   2011                                     The Artist       5           10   
9   2010                              The King's Speech       4           12   
10  2009                                The Hurt Locker       6            9   
11  2008                            Slumdog Millionaire       8           10   
12  2007                         No Country for Old Men       4            8   
13  2006                                   The Departed       4            5   
14  2005                                          Crash       3            6   
15  2004                            Million Dollar Baby       4            7   
16  2003  The Lord of the Rings: The Return of the King      11           11   
17  2002                                        Chicago       6           13   
18  2001                               A Beautiful Mind       4            8   
19  2000                                      Gladiator       5           12   
20  1999                                American Beauty       5            8   
21  1998                            Shakespeare in Love       7           13   
22  1997                                        Titanic      11           14   
23  1996                            The English Patient       9           12   
24  1995                                     Braveheart       5           10   
25  1994                                   Forrest Gump       6           13   
26  1993                               Schindler's List       7           12   
27  1992                                     Unforgiven       4            9   
28  1991                       The Silence of the Lambs       5            7   
29  1990                             Dances with Wolves       7           12   
30  1989                             Driving Miss Daisy       4            9   
31  1988                                       Rain Man       4            8   
32  1987                               The Last Emperor       9            9   
33  1986                                        Platoon       4            8   
34  1985                                  Out of Africa       7           11   
35  1984                                        Amadeus       8           11   
36  1983                            Terms of Endearment       5           11   
37  1982                                         Gandhi       8           11   
38  1981                               Chariots of Fire       4            7   
39  1980                                Ordinary People       4            6   
40  1979                              Kramer vs. Kramer       5            9   
41  1978                                The Deer Hunter       5            9   
42  1977                                     Annie Hall       4            5   
43  1976                                          Rocky       3           10   
44  1973                                      The Sting       7           10   
45  1972                                  The Godfather       3           10   
46  1971                          The French Connection       5            8   
47  1970                                         Patton       7           10   
48  1969                                Midnight Cowboy       3            7   
49  

**Eliminamos columnas que no nos interesan ahora**

In [187]:
df_movies_final = df_movies_final.drop(columns = ['Ratio Nominations/Awards', 'Runtime', 'Genre','Director','Actors','Country'])

**Cambiamos el orden de las columnas restantes para una mejor visibilidad de la información y conclusiones.**

In [189]:
df_movies_final = df_movies_final[['Year','Title','Awards','Nominations','imdbRating',]]
df_movies_final

Year                                          Title  Awards  Nominations  \
0   2019                                       Parasite       4            6   
1   2018                                     Green Book       3            5   
2   2017                             The Shape of Water       4           13   
3   2016                                      Moonlight       3            8   
4   2015                                      Spotlight       2            6   
5   2014                                        Birdman       4            9   
6   2013                               12 Years a Slave       3            9   
7   2012                                           Argo       3            7   
8   2011                                     The Artist       5           10   
9   2010                              The King's Speech       4           12   
10  2009                                The Hurt Locker       6            9   
11  2008                            Slumdog Millionaire       8           10   
12  2007                         No Country for Old Men       4            8   
13  2006                                   The Departed       4            5   
14  2005                                          Crash       3            6   
15  2004                            Million Dollar Baby       4            7   
16  2003  The Lord of the Rings: The Return of the King      11           11   
17  2002                                        Chicago       6           13   
18  2001                               A Beautiful Mind       4            8   
19  2000                                      Gladiator       5           12   
20  1999                                American Beauty       5            8   
21  1998                            Shakespeare in Love       7           13   
22  1997                                        Titanic      11           14   
23  1996                            The English Patient       9           12   
24  1995                                     Braveheart       5           10   
25  1994                                   Forrest Gump       6           13   
26  1993                               Schindler's List       7           12   
27  1992                                     Unforgiven       4            9   
28  1991                       The Silence of the Lambs       5            7   
29  1990                             Dances with Wolves       7           12   
30  1989                             Driving Miss Daisy       4            9   
31  1988                                       Rain Man       4            8   
32  1987                               The Last Emperor       9            9   
33  1986                                        Platoon       4            8   
34  1985                                  Out of Africa       7           11   
35  1984                                        Amadeus       8           11   
36  1983                            Terms of Endearment       5           11   
37  1982                                         Gandhi       8           11   
38  1981                               Chariots of Fire       4            7   
39  1980                                Ordinary People       4            6   
40  1979                              Kramer vs. Kramer       5            9   
41  1978                                The Deer Hunter       5            9   
42  1977                                     Annie Hall       4            5   
43  1976                                          Rocky       3           10   
44  1973                                      The Sting       7           10   
45  1972                                  The Godfather       3           10   
46  1971                          The French Connection       5            8   
47  1970                                         Patton       7           10   
48  1969                                Midnight Cowboy       3            7   
49  

**Vamos a consultar cuántos Oscar en total han ganado las 10 películas ganadoras "Oscar a la mejor película" con mayor puntuación en IMDB.**

In [190]:
df_movies_final_rating = df_movies_final.sort_values(('imdbRating'), ascending=False)
df_movies_final_rating.reset_index(drop=True).head(10)

Year                                          Title  Awards  Nominations  \
0  1972                                  The Godfather       3           10   
1  2003  The Lord of the Rings: The Return of the King      11           11   
2  1993                               Schindler's List       7           12   
3  1994                                   Forrest Gump       6           13   
4  1991                       The Silence of the Lambs       5            7   
5  2019                                       Parasite       4            6   
6  1943                                     Casablanca       3            8   
7  2006                                   The Departed       4            5   
8  2000                                      Gladiator       5           12   
9  1973                                      The Sting       7           10   

  imdbRating  
0        9.2  
1        8.9  
2        8.9  
3        8.8  
4        8.6  
5        8.6  
6        8.5  
7        8.5  
8        8.5  
9        8.3

Vemos por ejemplo como la película mejor puntuada es The GodFather, con 9.2. Sin embargo sólo ganó 3 Oscars.

Por otro lado, de las mejores puntuadas la que más Oscars ha ganado ha sido The Lord of The Rings, con 11 Oscars.

**Vamos a consultar cuál es la puntuación en IMDB de las películas que más Oscars han recibido**

In [191]:
df_movies_final_awards = df_movies_final.sort_values(('Awards'), ascending=False)
df_movies_final_awards.reset_index(drop=True).head(10)

Year                                          Title  Awards  Nominations  \
0  1997                                        Titanic      11           14   
1  2003  The Lord of the Rings: The Return of the King      11           11   
2  1959                                        Ben-Hur      11           12   
3  1961                                West Side Story      10           11   
4  1939                             Gone with the Wind      10           13   
5  1996                            The English Patient       9           12   
6  1987                               The Last Emperor       9            9   
7  1958                                           Gigi       9            9   
8  1984                                        Amadeus       8           11   
9  1982                                         Gandhi       8           11   

  imdbRating  
0        7.8  
1        8.9  
2        8.1  
3        7.5  
4        8.1  
5        7.4  
6        7.7  
7        6.7  
8        8.3  
9        8.0

Vemos por ejemplo que de las películas que han ganado 10 Oscars o más, la peor puntuada es West Side Story.

Y por ejemplo que Titanic, con el mayor número de Oscars y Nominaciones, no llega al 8 de puntuación.

Por otro lado, la peor puntuada de las 10 películas con más Oscars sería Gigi con 6.7 de puntuación.
